tools/configure/cmdArgs.ml

```diff
@@ -0,0 +1,199 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+open Util
+
+(** * Command-line parsing *)
+
+type ide = Opt | Byte | No
+
+type nativecompiler = NativeYes | NativeNo | NativeOndemand
+
+module Prefs = struct
+
+type t = {
+  prefix : string option;
+  interactive : bool;
+  libdir : string option;
+  configdir : string option;
+  datadir : string option;
+  mandir : string option;
+  docdir : string option;
+  arch : string option;
+  natdynlink : bool;
+  coqide : ide option;
+  macintegration : bool;
+  browser : string option;
+  withdoc : bool;
+  byteonly : bool;
+  bin_annot : bool;
+  annot : bool;
+  bytecodecompiler : bool;
+  nativecompiler : nativecompiler;
+  coqwebsite : string;
+  warn_error : bool;
+  dune_profile : string;
+  install_enabled : bool;
+}
```

Comment:
Can we get some documentation on each of these fields? Or at least a comment saying that it's use is explained in `args_options` later?

Reply:
Good idea.

Reply:
Done, but in the .mli file.